File: templates/src/main/resources/templates/mod/lang/java/src/main/java/{{packagePath}}/{{class}}.java
package {{package}};

import dev.ancaria.coderpack.api.SacredMod;
import dev.ancaria.coderpack.api.Subscribe;
import dev.ancaria.coderpack.api.event.Hero;

/** {{description}} */
public final class {{class}} extends SacredMod {

    @Override
    public void onLoad() {
        // Every public @Subscribe method on this object becomes a listener.
        getContext().getRegistry().getEventRegistry().register(this);
        getContext().log("Loaded.");
    }

    /**
     * The hero was found: once per world load, and again when the player
     * switches character. Nothing before this can touch the player, because
     * before this there is no world.
     */
    @Subscribe
    public void onHero(Hero hero) {
        getContext().log("Hero reached level " + hero.getLevel() + ".");
    }
}

File: templates/src/main/resources/templates/mod/lang/kotlin/src/main/kotlin/{{packagePath}}/{{class}}.kt
package {{package}}

import dev.ancaria.coderpack.api.SacredMod
import dev.ancaria.coderpack.api.event.Hero

/** {{description}} */
class {{class}} : SacredMod() {

    override fun onLoad() {
        // The hero was found: once per world load, and again when the player
        // switches character. Nothing before this can touch the player,
        // because before this there is no world.
        context.registry.eventRegistry.on(Hero::class.java) { hero ->
            context.log("Hero reached level ${hero.level}.")
        }
        context.log("Loaded.")
    }
}
